app/src/main/java/com/raulh82vlc/functionalkotlinarrow/data/network/error/ApiException.kt

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2018 Raul Hernandez Lopez
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.raulh82vlc.functionalkotlinarrow.data.network.error
+
+class ApiException(msg: String, cause: Throwable?) : Exception(msg, cause)
```

Comment:
Rather than use exceptions, it may be better to have a closed hierarchy of error types (seal class) with the possible errors. Also means that we don't have to pass around Exceptions.

Reply:
This is a total refactor though, I'll keep it in mind for the next round then